Bottom: b1bb4ff2d433afbfacb18cfacdd814e3ae7e931c
Top:    8ed9a7ed5ec760ca6e9e2a3e6b531402eb1219fd
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 16:31:26 +0000

Enable output of via pads, and dial down circular approximation accuracy


---

diff --git a/src/hid/common/object3d.c b/src/hid/common/object3d.c
index c25a9b7..dc59659 100644
--- a/src/hid/common/object3d.c
+++ b/src/hid/common/object3d.c
@@ -1147,11 +1147,11 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
     END_LOOP;
 
     fprintf (stderr, "Accumulating pin + via pads\n");
-//    r_search (PCB->Data->pin_tree, &bounds, NULL, pv_copper_callback, &info);
-//    r_search (PCB->Data->via_tree, &bounds, NULL, pv_copper_callback, &info);
+    r_search (PCB->Data->pin_tree, &bounds, NULL, pv_copper_callback, &info);
+    r_search (PCB->Data->via_tree, &bounds, NULL, pv_copper_callback, &info);
 #endif
 
-#if 0
+#if 1
     if (group == top_group ||
         group == bottom_group)
       {
@@ -1161,6 +1161,31 @@ object3d_from_copper_layers_within_area (POLYAREA *area)
       }
 #endif
 
+  /* TODO: Inter-layer features
+   *
+   * Accumulate a circular polygon for each plated hole we may cut, ensuring
+   * the finished polygon contour extends to include the via barrel extents.
+   *
+   * Subtract non-plated hole contours from the polygons.
+   *
+   * For each hole, add the via-barrel between layers... removing the Object3D
+   * from the list of objects as they become joined with some other. (The final
+   * list of objects shuold match 1:1 with resultant bodies, and contain no
+   * duplicates.
+   *
+   * To accomodate overlapping drill holes, accumulate all via-barrels into a
+   * polygon, and subtract that from the positive copper polygon. As we already
+   * added via-barrels to each copper layer, the each barrel extrusion contour
+   * should match only one body of copper on a given layer.
+   *
+   * Remove the drilled hole down each via by extruding the additional faces.
+   *
+   * To accomodate overlapping drill holes, accumulate all drills into a polygon,
+   * and subtract that from the positive copper polygon. Any subtracted contour
+   * should at this point match to one body of copper on a given layer.
+   *
+   */
+
   if (info.poly == NULL)
     {
       fprintf (stderr, "Skipping layer group %i, info.poly was NULL\n", group);
